pwndbg/commands/chain.py

```diff
@@ -0,0 +1,394 @@
+        same as outer structure.
+    3 - Next pointer is field of inner nested structure, pointed to type is the
+        same as inner structure.
+Types 2 and 3 require --inner <inner-name> to be specified.
+
+Example 1:
+
+```
+struct node {
+    int value;
+    struct node *next;
+};
+struct node node_c = { 2, NULL };
+struct node node_b = { 1, &node_c };
+struct node node_a = { 0, &node_b };
+```
+
+pwndbg> chain node_a next
+0x4000011050 <node_a>: {
+  value = 0,
+  next = 0x4000011040 <node_b>
+}
+0x4000011040 <node_b>: {
+  value = 1,
+  next = 0x4000011010 <node_c>
+}
+0x4000011010 <node_c>: {
+  value = 2,
+  next = 0x0
+}
+
+Example 2:
+
+```
+struct node_inner_a {
+    struct node_inner_a *next;
+};
+struct inner_a_node {
+    int value;
+    struct node_inner_a inner;
+};
+struct inner_a_node inner_a_node_c = { 2, { NULL } };
+struct inner_a_node inner_a_node_b = { 1, { &inner_a_node_c.inner } };
+struct inner_a_node inner_a_node_a = { 0, { &inner_a_node_b.inner } };
+```
+
+pwndbg> chain inner_a_node_a -i inner next
+0x4000011070 <inner_a_node_a>: {
+  value = 0,
+  inner = {
+    next = 0x4000011068 <inner_a_node_b+8>
+  }
+}
+0x4000011060 <inner_a_node_b>: {
+  value = 1,
+  inner = {
+    next = 0x4000011028 <inner_a_node_c+8>
+  }
+}
+0x4000011020 <inner_a_node_c>: {
+  value = 2,
+  inner = {
+    next = 0x0
+  }
+}
+
+Example 3:
+
+```
+struct inner_b_node;
+struct node_inner_b {
+    struct inner_b_node *next;
+};
+struct inner_b_node {
+    int value;
+    struct node_inner_b inner;
+};
+struct inner_b_node inner_b_node_c = { 2, { NULL } };
+struct inner_b_node inner_b_node_b = { 1, { &inner_b_node_c } };
+struct inner_b_node inner_b_node_a = { 0, { &inner_b_node_b } };
+```
+
+pwndbg> chain inner_b_node_a -i inner next
+0x4000011090 <inner_b_node_a>: {
+  value = 0,
+  inner = {
+    next = 0x4000011080 <inner_b_node_b>
+  }
+}
+0x4000011080 <inner_b_node_b>: {
+  value = 1,
+  inner = {
+    next = 0x4000011030 <inner_b_node_c>
+  }
+}
+0x4000011030 <inner_b_node_c>: {
+  value = 2,
+  inner = {
+    next = 0x0
+  }
+}
+
+""",
+)
+parser.add_argument(
+    "path",
+    type=str,
+    help="The first element of the chain",
+)
+parser.add_argument(
+    "next", type=str, help="The name of the field pointing to the next element in the chain"
+)
+parser.add_argument(
+    "-s",
+    "--sentinel",
+    dest="sentinel",
+    type=int,
+    default=0,
+    help="The address that stands in for an end of list marker in a non-cyclic list",
+)
+parser.add_argument(
+    "-i",
+    "--inner",
+    dest="inner_name",
+    type=str,
+    help="The name of the inner nested structure where the next pointer is stored",
+)
+parser.add_argument(
+    "-f",
+    "--field",
+    dest="field_name",
+    type=str,
+    help="The name of the field to be displayed, if only one is desired",
+)
+
+
+@pwndbg.commands.ArgparsedCommand(parser, command_name="chain")
+def chain_cmd(path, next, sentinel, inner_name, field_name) -> None:
```

Comment:
Is there any reason we don't just name the function `chain`? :)

```suggestion
def chain(path, next, sentinel, inner_name, field_name) -> None:
```

Reply:
Nope. :)
And fixed.